rapid7/metasploit-framework
# Add VMware vRealize Log Insight unauthenticated RCE exploit

Hello :wave:

This module exploits multiple vulnerabilities for achieving unauthenticated remote code execution on the VMware vRealize Log Insight version `v8.x`. Module achieves code execution via triggering a `RemotePakDownloadCommand` command via the exposed thrift service after obtaining the node token by calling a `GetConfigRequest` thrift command. After the download, we trigger a `PakUpgradeCommand` for extracting the specially crafted TAR archive that we served, which then will place the JSP payload under a certain API endpoint (pre-authenticated) location for us to call.


I'm aware that the code looks ugly, but since this is an important target, I wanted to push it asap. I had to manually construct thrift packages because `Rex::Proto::Thrift` is very premature at the moment. Also, I couldn't find a better way to embed the file contents. (certs and checksums) Open to suggestions.

## Testing Environment Setup
For installing the vulnerable version follow the steps below,
1. To obtain the vulnerable OVA image, first create a customer account at [VMware](https://customerconnect.vmware.com/account-registration) (trial license is sufficient)
2. Navigate [here](https://customerconnect.vmware.com/account-registration) and download `Virtual Appliance`
3. Import the OVA image into a virtualization software (VirtualBox is used for this case).
4. Start the `VMware_vCenter_Log_Insight` image and proceed with the initial installation steps through the web interface of the product.

After these steps, the web portal (port 80/443) and Apache thrift service (port 16520) should be accessible.

## Verification
List the steps needed to make sure this thing works

- [ ] Do `use exploit/linux/http/vmware_vrli_rce`
- [ ] Do `set RHOST [IP]`
- [ ] Do `check`
